client/lxd_images.go

```diff
@@ -484,6 +484,14 @@ func (r *ProtocolLXD) CreateImage(image api.ImagesPost, args *ImageCreateArgs) (
 		req.Header.Set("User-Agent", r.httpUserAgent)
 	}
 
+	if image.Source != nil && image.Source.Fingerprint != "" && image.Source.Secret != "" {
```

Comment:
If we introduce `Mode`, we'll want this to check for `image.Source.Mode == "push"` too